### PR TITLE
wrap registration block to catch error

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Fixes issue when running against an older mattermost-server that does not implement `registry.registerChannelIntroButtonAction`

#### Ticket Link
https://github.com/mattermost/focalboard/issues/2251

### CODE REVIEW

**Review by harshilsharma63 — APPROVED** [review]
This works for now but can we update this to match the existing pattern? Currently, we check if `this.registry.<some hook>` is truthy, and use it if so.

For example this line https://github.com/mattermost/focalboard/blob/9d12efa2a2577336c25cfc7a99e6bcc11f860106/mattermost-plugin/webapp/src/index.tsx#L135

File: mattermost-plugin/webapp/src/index.tsx [hunk 1]
```diff
@@ -148,7 +148,12 @@ export default class Plugin {
                 UserSettings.lastViewId = null
                 window.open(`${windowAny.frontendBaseURL}/workspace/${currentChannel}`, '_blank', 'noopener')
             }
-            this.channelHeaderButtonId = registry.registerChannelIntroButtonAction(<FocalboardIcon/>, goToFocalboardTemplate, 'Boards')
+            try {
```
Comment (jespino) [inline mattermost-plugin/webapp/src/index.tsx:151]: I think I prefer a this way:
```ts
// backward compatiblity for version < v6.5
if (registry.registerChannelIntroButtonAction) {
    this.channelHeaderButtonId = registry.registerChannelIntroButtonAction(<FocalboardIcon/>, goToFocalboardTemplate, 'Boards')
}
```

**Review by jespino — CHANGES_REQUESTED** [review]
I think I prefer a different way that doesn't mask possible unrelated errors.

**Pull request merged**